src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1,855 +1,245 @@
+    import type { APIContext } from 'astro';
+
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' }, props: { author: 'Blu' } },
+        { params: { id: '2' }, props: { author: 'Erika' } },
+        { params: { id: '3' }, props: { author: 'Matthew' } }
+      ];
+    }
+
+    export function GET({ props }: APIContext) {
+      return new Response(
+        JSON.stringify({ author: props.author }),
+      );
+    }
+    ```
+  </TabItem>
+
+</Tabs>
+See also: [Data Passing with `props`](/en/reference/routing-reference/#data-passing-with-props)
 
-`Astro.slots.render()` optionally accepts a second argument: an array of parameters that will be forwarded to any function children. This can be useful for custom utility components.
-
-For example, this `<Shout />` component converts its `message` prop to uppercase and passes it to the default slot:
+### `params`
 
-```astro title="src/components/Shout.astro" "await Astro.slots.render('default', [message])"
+`params` is an object containing the values of dynamic route segments matched for a route request. Its keys must match the [dynamic route parameters](/en/guides/routing/#dynamic-routes) from the page file or endpoint route.
+
+In static builds, this will be the `params` returned by `getStaticPaths()` used for prerendering [dynamic routes](/en/guides/routing/#dynamic-routes):
+
+<Tabs>
+  <TabItem label="Astro.params">
+    ```astro title="src/pages/posts/[id].astro" {9} "params:"
+    ---
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' } },
+        { params: { id: '2' } },
+        { params: { id: '3' } }
+      ];
+    }
+    const { id } = Astro.params;
+    ---
+    <h1>{id}</h1>
+    ```
+  </TabItem>
+  <TabItem label="context.params">
+    ```ts title="src/pages/posts/[id].json.ts" "params:" {11-14}
+    import type { APIContext } from 'astro';
+
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' } },
+        { params: { id: '2' } },
+        { params: { id: '3' } }
+      ];
+    }
+
+    export function GET({ params }: APIContext) {
+      return new Response(
+        JSON.stringify({ id: params.id }),
+      );
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+
+When pages are rendered on demand, `params` can be any value matching the path segments in the dynamic route pattern.
+
+```astro title="src/pages/posts/[id].astro" "Astro.params"
 ---
-const message = Astro.props.message.toUpperCase();
-let html = '';
-if (Astro.slots.has('default')) {
-  html = await Astro.slots.render('default', [message]);
-}
----
-<Fragment set:html={html} />
-```
+import { getPost } from '../api';
 
-A callback function passed as `<Shout />`’s child will receive the all-caps `message` parameter:
+const post = await getPost(Astro.params.id);
 
-```astro title="src/pages/index.astro"
----
-import Shout from "../components/Shout.astro";
+// No posts found with this ID
+if (!post) {
+  Astro.redirect("/404")
```

Comment:
It's time to fix my error... 👀 When I updated this part, I forgot the return statement: `Astro.redirect` should be returned!
```suggestion
  return Astro.redirect("/404")
```